Avahi is a non-optional component

Avahi is required to generate the list of available RTSP streams. This is supposed to be an optional feature and component.

However following the default instructions seem to imply it is required.
```
$ ./autogen.sh && ./configure
```
I tried 
```
make
```
This fails with error:

```In file included from src/stream_builder.h:19:0,
                 from src/stream_builder.cpp:21:
src/stream_manager.h:20:32: fatal error: avahi-common/watch.h: No such file or directory
```

So unless ```make``` in bare format is not expected, avahi is required.

--- a/Makefile.am
+++ b/Makefile.am
@@ -291,8 +291,6 @@ samples_deps = samples_camera_sample_mavlink_client_LDADD
 endif
 
 if ENABLE_GAZEBO
-AM_CFLAGS += \
-        ${GZB_CFLAGS}
 
 AM_CXXFLAGS += \
         ${GZB_CFLAGS}

--- a/src/CameraDeviceV4l2.cpp
+++ b/src/CameraDeviceV4l2.cpp
@@ -150,7 +150,7 @@ int CameraDeviceV4l2::resetParams(CameraParameters &camParam)
             if (queryctrl.flags & V4L2_CTRL_FLAG_DISABLED)
                 continue;
         }
-        int ret = v4l2_set_control(fd, queryctrl.id, queryctrl.default_value);
+        ret = v4l2_set_control(fd, queryctrl.id, queryctrl.default_value);
         //    log_debug("return value for:%s : %d : %d",queryctrl.name,queryctrl.id,ret);
 
         switch (queryctrl.id) {

--- a/src/mavlink_server.cpp
+++ b/src/mavlink_server.cpp
@@ -16,6 +16,7 @@
  * limitations under the License.
  */
 #include <assert.h>
+#include <cmath>
 #include <mavlink.h>
 #include <sys/socket.h>
 #include <unistd.h>
@@ -31,6 +32,8 @@ using namespace std::placeholders;
 #define DEFAULT_MAVLINK_BROADCAST_ADDR "255.255.255.255"
 #define DEFAULT_RTSP_SERVER_ADDR "0.0.0.0"
 #define MAX_MAVLINK_MESSAGE_SIZE 1024
+
+static const float epsilon = std::numeric_limits<float>::epsilon();
 
 MavlinkServer::MavlinkServer(ConfFile &conf, std::vector<std::unique_ptr<Stream>> &streams,
                              RTSPServer &rtsp)
@@ -115,8 +118,9 @@ void MavlinkServer::_handle_request_camera_information(const struct sockaddr_in 
     mavlink_message_t msg;
     bool success = false;
 
-    /* Check for Request camera capabilities flag*/
-    if (cmd.param1 != 1) {
+    // Take no action if flag not set
+    if (std::abs(cmd.param1) <= epsilon) {
+        log_warning("No Action");
         _send_ack(addr, cmd.command, cmd.target_component, true);
         return;
     }
@@ -147,8 +151,9 @@ void MavlinkServer::_handle_request_camera_settings(const struct sockaddr_in &ad
 {
     log_debug("%s", __func__);
 
-    // Check for Request camera settings flag
-    if (cmd.param1 != 1) {
+    // Take no action if flag not set
+    if (std::abs(cmd.param1) <= epsilon) {
+        log_warning("No Action");
         _send_ack(addr, cmd.command, cmd.target_component, true);
         return;
     }
@@ -178,8 +183,9 @@ void MavlinkServer::_handle_request_storage_information(const struct sockaddr_in
 {
     log_debug("%s", __func__);
 
-    // Check for Request storage information flag
-    if (cmd.param2 != 1) {
+    // Take no action if flag not set
+    if (std::abs(cmd.param2) <= epsilon) {
+        log_warning("No Action");
         _send_ack(addr, cmd.command, cmd.target_component, true);
         return;
     }
@@ -285,8 +291,9 @@ void MavlinkServer::_handle_request_camera_capture_status(const struct sockaddr_
 {
     log_debug("%s", __func__);
 
-    // Check for Request camera capture status flag
-    if (cmd.param1 != 1) {
+    // Take no action if flag not set
+    if (std::abs(cmd.param1) <= epsilon) {
+        log_warning("No Action");
         _send_ack(addr, cmd.command, cmd.target_component, true);
         return;
     }
@@ -519,8 +526,11 @@ void MavlinkServer::_handle_param_ext_set(const struct sockaddr_in &addr, mavlin
 void MavlinkServer::_handle_reset_camera_settings(const struct sockaddr_in &addr,
                                                   mavlink_command_long_t &cmd)
 {
+    log_debug("%s", __func__);
 
-    if (cmd.param1 != 1) {
+    // Take no action if flag not set
+    if (std::abs(cmd.param1) <= epsilon) {
+        log_warning("No Action");
         _send_ack(addr, cmd.command, cmd.target_component, true);
         return;
     }

--- a/src/stream_manager.h
+++ b/src/stream_manager.h
@@ -17,13 +17,13 @@
  */
 #pragma once
 
-#include <avahi-common/watch.h>
 #include <memory>
 #include <string>
 #include <vector>
 
 #ifdef ENABLE_AVAHI
 #include "avahi_publisher.h"
+#include <avahi-common/watch.h>
 #endif
 #include "conf_file.h"
 #ifdef ENABLE_MAVLINK

